commerce-service/src/main/java/com/liferay/commerce/internal/order/term/contributor/CommerceRecipientDefinitionTermContributor.java

```diff
@@ -0,0 +1,247 @@
+
+		if (term.equals(_ORDER_CREATOR)) {
+			User user = _userLocalService.getUser(commerceOrder.getUserId());
+
+			return String.valueOf(user.getUserId());
+		}
+
+		if (term.equals(_ACCOUNT_ROLE_ADMINISTRATOR)) {
+			CommerceAccount commerceAccount =
+				commerceOrder.getCommerceAccount();
+
+			Role accountAdminRole = _roleLocalService.getRole(
+				commerceOrder.getCompanyId(),
+				CommerceAccountConstants.ACCOUNT_ADMINISTRATOR_ROLE_NAME);
+
+			return _getUserIds(commerceAccount, accountAdminRole);
+		}
+
+		if (term.equals(_ACCOUNT_ROLE_ORDER_MANAGER)) {
+			CommerceAccount commerceAccount =
+				commerceOrder.getCommerceAccount();
+
+			Role accountAdminRole = _roleLocalService.getRole(
+				commerceOrder.getCompanyId(), "Order Manager");
+
+			return _getUserIds(commerceAccount, accountAdminRole);
+		}
+
+		if (term.startsWith("%USER_GROUP_")) {
+			String[] s = term.split("_");
+
+			String userGroupName = StringUtil.removeChar(s[2], '%');
+
+			UserGroup userGroup = _userGroupLocalService.getUserGroup(
+				commerceOrder.getCompanyId(), userGroupName);
+
+			return _getUserIds(userGroup);
+		}
+
+		return term;
+	}
+
+	@Override
+	public String getLabel(String term, Locale locale) {
+		ResourceBundle resourceBundle = ResourceBundleUtil.getBundle(
+			"content.Language", locale, getClass());
+
+		return LanguageUtil.get(
+			resourceBundle, _commerceOrderDefinitionTermsMap.get(term));
+	}
+
+	@Override
+	public List<String> getTerms() {
+		return new ArrayList<>(_commerceOrderDefinitionTermsMap.keySet());
+	}
+
+	private String _getUserIds(
+			CommerceAccount commerceAccount, Role accountAdminRole)
+		throws PortalException {
+
+		List<CommerceAccountUserRel> commerceAccountUserRels =
+			_commerceAccountUserRelLocalService.getCommerceAccountUserRels(
+				commerceAccount.getCommerceAccountId());
+
+		StringBundler resultsSB = new StringBundler();
+
+		for (CommerceAccountUserRel commerceAccountUserRel :
+				commerceAccountUserRels) {
+
+			User user = commerceAccountUserRel.getUser();
+
+			List<Role> userRoles = _roleLocalService.getUserRoles(
+				user.getUserId());
+
+			if (userRoles.contains(accountAdminRole)) {
+				resultsSB.append(user.getUserId());
+				resultsSB.append(",");
+			}
+		}
+
+		return resultsSB.toString();
+	}
+
+	private String _getUserIds(UserGroup userGroup) throws PortalException {
+		List<User> groupUsers = _userLocalService.getGroupUsers(
+			userGroup.getGroupId());
+
+		StringBundler results = new StringBundler();
+
+		for (User user : groupUsers) {
+			results.append(user.getUserId());
+			results.append(",");
+		}
+
+		return results.toString();
+	}
+
+	private static final String _ACCOUNT_ROLE_ADMINISTRATOR =
+		"[%ACCOUNT_ROLE_ADMINISTRATOR%]";
+
+	private static final String _ACCOUNT_ROLE_ORDER_MANAGER =
+		"[%ACCOUNT_ROLE_ORDER_MANAGER%]";
+
+	private static final String _ORDER_CREATOR = "[%ORDER_CREATOR%]";
+
+	private static final String _USER_GROUP =
+		"[%USER_GROUP_NAMEOFTHEUSERGROUP%]";
```

Comment:
Could we call this `[%USER_GROUP_NAME%]` instead? Seems more... standard. Seems like it maybe makes sense to change the constant name to `_USER_GROUP_NAME` as well, if possible.

Reply:
True, I'll change it.
But we need it to be clear because in order to work we need something like
[%USER_GROUP_Gold%]
[%USER_GROUP_Silver%]

With only [%USER_GROUP_NAME%] I think people will write something like
[%Gold%]
[%Silver%]

That will not work